react-sizeme.d.ts

```diff
@@ -0,0 +1,26 @@
+declare module 'react-sizeme' {
+
+  type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
+
+  export interface SizeMeProps {
+    size: {
+      width: number | null;
+      height: number | null;
+    };
+  }
+
+  export interface SizeMeOptions {
+    monitorWidth?: boolean;
+    monitorHeight?: boolean;
+    monitorPosition?: boolean;
+    refreshRate?: number;
+    refreshMode?: 'throttle' | 'debounce';
+    noPlaceholder?: boolean;
+  }
+
+  export class SizeMe extends React.Component<SizeMeOptions> {
```

Comment:
I think we should revise this typing to take into account it's render props based API.

I am not familiar with Typescript but found the following possibly helpful article:

https://medium.com/@jrwebdev/react-render-props-in-typescript-b561b00bc67c

Reply:
I believe from reading the article that this typing was missing on the `SizeMeOptions` interface: 

    children(props: SizeMeProps): JSX.Element;

Now TS complains if we pass anything but a function as a children of the `<SizeMe>` component, and we get the typings of `SizeMeProps` on that function as well. (Before on my code I had to manually specify `{size}` was `SizeMeProps` like this `<SizeMe>{({size}: SizeMeProps) => ... </SizeMe>`.

Also, I believe `SizeMeProps` interface doesn't have to be exported anymore.

Reply:
Ok, nice one, sounds good. Did you commit/push your latest? Not seeing these changes. 😀

Reply:
Great! I'll commit that change now that you agreed with it.